io/disk/iozone.py

```diff
@@ -0,0 +1,577 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Harish <harisrir@linux.vnet.ibm.com>
+
+# Based on code by:
+#   author: Ying Tao <yingtao@cn.ibm.com>
+#   author: Lucas Meneghel Rodrigues <lmr@redhat.com>
+#   copyright: 2006 IBM
+#   copyright: 2008 Red Hat, Inc.
+
+import os
+import re
+import json
+import logging
+
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import distro
+from avocado.utils import data_structures
+from avocado.utils import astring
+from avocado.utils.software_manager import SoftwareManager
+
+
+_LABELS = ['file_size', 'record_size', 'write', 'rewrite', 'read', 'reread',
+           'randread', 'randwrite', 'bkwdread', 'recordrewrite', 'strideread',
+           'fwrite', 'frewrite', 'fread', 'freread']
+
+
+class IOzoneAnalyzer(object):
+
+    """
+    Analyze an unprocessed IOzone file, and generate the following types of
+    report:
+
+    * Summary of throughput for all file and record sizes combined
+    * Summary of throughput for all file sizes
+    * Summary of throughput for all record sizes
+
+    If more than one file is provided to the analyzer object, a comparison
+    between the two runs is made, searching for regressions in performance.
+    """
+
+    def __init__(self, list_files, output_dir):
+        self.list_files = list_files
+        if not os.path.isdir(output_dir):
+            os.makedirs(output_dir)
+        self.output_dir = output_dir
+        logging.info("Results will be stored in %s", output_dir)
```

Comment:
Oh, this whole test uses the `root` python logger, while tests should use the `avocado.test` logger instead. The recommended way is to use the log from `Test.log` as it contains additional features to detect `warn` usage.

If you don't want to pass the log from the test itself, you could also use `logging.getLogger("avocado.test")`, but passing the log from test would be preferred.